feat: Add constants.humanName to allow countries to have custom full name format

Siyasanga opened the pull request:
In order to allow countries to define their own full name format, we are adding a constants.humanName into the client.csv translations, which will then be used everywhere where we are displaying system users and citizen's name in the systems.

Addresses https://github.com/opencrvs/opencrvs-core/issues/6830

Countryconfig: https://github.com/opencrvs/opencrvs-countryconfig/pull/314
Farajaland: https://github.com/opencrvs/opencrvs-farajaland/pull/1164

Siyasanga committed:
Add humanName to constants.ts

To allow countries to have custom ordering for full names

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Use constants.humanName on the UserList.tsx comp

To ensure that we get the format from the country-config

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Create a utility function getLocalisedname()

To make the name more usable we had to extract the name formatting logic into it's own function.

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Use getLocalisedName on the UserList component

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Use getLocalisedName on the UserAudit comp

Replace older logic to get the name which was based on an assumption that we support names in multiple languages

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
refactor: use getLocalisedName() @ InProgress.tsx

We've found cleaner way to make the rendered name customizable for each country through client copy from country-config

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Refactor the work queues to use getLocalisedName()

We need to update all the places where a citizen's name is being referenced to show it in the format that the country chooses

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Record changes in the CHANGELOG for this PR

https://github.com/opencrvs/opencrvs-core/issues/6830

Zangetsu101 reviewed (changes requested):
@Siyasanga After taking a look at the changes made to the e2e tests, I would say it's better to change all those places in core to use the new format instead. As if only some of the places use the new format, then it would just create confusion

Siyasanga committed:
Use humanName for other search/transformer.ts

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Fix type error 4 the findSavedReference()

Siyasanga committed:
Fix faling unit test after using humanName

The expected name should match the format for the given country.

https://github.com/opencrvs/opencrvs-core/issues/6830

Zangetsu101 commented on packages/client/src/utils/draftUtils.test.ts:
```diff
@@ -17,29 +17,49 @@ import type {
+  },
+  cache
+)
+const intlBangla = createIntl(
+  {
+    locale: 'en',
+    messages: {}
+  },
+  cache
+)
```
Is this supposed to be like so?

Siyasanga replied:
Nice catch, thanks Tameem. Fixed

Siyasanga committed:
Add the correct local for intlBangla

https://github.com/opencrvs/opencrvs-core/issues/6830

Siyasanga committed:
Pass intl object for myDrafts transformDraftContent

https://github.com/opencrvs/opencrvs-core/issues/6830

euanmillar commented:
@Zangetsu101 @Siyasanga and I dont know why knip is failing here but tests are all passing and think this is ready to merge?

Zangetsu101 commented:
@Siyasanga What's the status of e2e tests? As long as they are passing, we can merge this in

Siyasanga committed:
Merge branch 'develop' into ocrvs-6830-human-name-constant

Siyasanga commented:
> @Siyasanga What's the status of e2e tests? As long as they are passing, we can merge this in

They are passing so I will merge this and the related [contry-config PR](https://github.com/opencrvs/opencrvs-countryconfig/pull/314)

Zangetsu101 commented on CHANGELOG.md:
```diff
@@ -29,6 +29,8 @@
 - A new GraphQL mutation `upsertRegistrationIdentifier` is added to allow updating the patient identifiers of a registration record such as NID [#8034](https://github.com/opencrvs/opencrvs-core/pull/8034)
 - A new GraphQL mutation `updateField` is added to allow updating any field in a record [#8291](https://github.com/opencrvs/opencrvs-core/pull/8291)
 - Updated GraphQL mutation `confirmRegistration` to allow adding a `comment` for record audit [#8197](https://github.com/opencrvs/opencrvs-core/pull/8197)
+- Introduced a new customisable UI component: Banner [#8276](https://github.com/opencrvs/opencrvs-core/issues/8276)
```
```suggestion
```

Zangetsu101 replied:
Seems to be something irrelevant
